locallib.php

```diff
@@ -2246,18 +2246,370 @@ function bigbluebuttonbn_output_recording_table($bbbsession, $recordings, $tools
+        $row->cells[] = $rowdata->preview;
+    }
+    $row->cells[] = $rowdata->date_formatted;
+    $row->cells[] = $rowdata->duration_formatted;
+    if ($bbbsession['managerecordings']) {
+        $row->cells[] = $rowdata->actionbar;
+    }
+    return $row;
+}
+
+/**
+ * Helper function builds element with actual images used in Opencast recording preview row based on a selected playback.
+ *
+ * @param string $imageurl
+ *
+ * @return string
+ */
+function bigbluebuttonbn_get_opencast_recording_data_row_preview_images($imageurl) {
+    global $CFG;
+    $recordingpreview  = html_writer::start_tag('div', array('class' => 'container-fluid'));
+    $recordingpreview .= html_writer::start_tag('div', array('class' => 'row'));
+    $recordingpreview .= html_writer::start_tag('div', array('class' => ''));
+    $recordingpreview .= html_writer::empty_tag(
+        'img',
+        array('src' => trim($imageurl) . '?' . time(), 'class' => 'recording-thumbnail pull-left')
+    );
+    $recordingpreview .= html_writer::end_tag('div');
+    $recordingpreview .= html_writer::end_tag('div');
+    $recordingpreview .= html_writer::start_tag('div', array('class' => 'row'));
+    $recordingpreview .= html_writer::tag(
+        'div',
+        get_string('view_recording_preview_help', 'bigbluebuttonbn'),
+        array('class' => 'text-center text-muted small')
+    );
+    $recordingpreview .= html_writer::end_tag('div');
+    $recordingpreview .= html_writer::end_tag('div');
+    return $recordingpreview;
+}
+
+/**
+ * Helper function to get BBB recordings from the Opencast video avaialble in the course.
+ * It uses block_opencast for getting all videos for the course and match them with meeting id.
+ * It uses tool_opencast for making an api call to get mediapackage of vidoes.
+ *
+ * @param object $bbbsession
+ * @param string $seriesid
+ *
+ */
+function bigbluebutton_get_opencast_recordings_for_table_view($bbbsession, $seriesid) {
+    $bbbocvideos = array();
+    // Initializing the api from tool_opencast plugin.
+    $api = new \tool_opencast\local\api();
+    // Getting an instance of apibridge from block_opencast plugin.
+    $opencast = \block_opencast\local\apibridge::get_instance();
+    // Getting the course videos from block_opencast plugin.
+    $ocvideos = $opencast->get_course_videos($bbbsession['course']->id);
+    if ($ocvideos->videos && !empty($ocvideos->videos)) {
+        foreach ($ocvideos->videos as $ocvideo) {
+            // Check subjects of opencast video contains $bbbsession['meetingid'].
+            if (in_array($bbbsession['meetingid'], $ocvideo->subjects)) {
+                // Converting $ocvideo object to array.
+                $ocvideoarray = json_decode(json_encode($ocvideo), true);
+                // Get mediapackage json using api call.
+                $url = '/search/episode.json?id=' . $ocvideo->identifier;
+                $search_result = json_decode($api->oc_get($url), true);
```

Comment:
I prefer to not to extend other plugins just for this integration to avoid complications.
I would suggest this way because it offer more control over what needs to be done here.